packages/popup/src/hooks/index.js

```diff
@@ -64,33 +82,229 @@ export const useFontSize = (
   }, [targetRef, hiddenRef, maxWidth, value, initialFontSize])
 }
 
-export const useEstimateTx = (tx = {}) => {
+export const useEstimateTx = (tx = {}, tokensAmount = {}) => {
   const {provider} = useRPCProvider()
   const currentNetwork = useCurrentNetwork() || {type: NETWORK_TYPE.CFX}
   const {type} = currentNetwork
-  const {from, to, value, data, nonce} = tx
+  const {from, to, value, data, nonce, gasPrice, gas, storageLimit} = tx
   const {
     value: rst,
     loading,
     error,
   } = useAsync(async () => {
-    if (!provider || !currentNetwork?.netId) return
+    if (!provider || !currentNetwork?.netId || (!to && !data)) return
     return await estimate(tx, {
       type,
       request: provider.request.bind(provider),
+      tokensAmount,
       isFluentRequest: true,
-      networkId: currentNetwork.netId,
+      // networkId: currentNetwork.netId,
     })
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [from, to, value, data, nonce, currentNetwork.netId])
+  }, [
+    from,
+    to,
+    value,
+    data,
+    nonce,
+    gasPrice,
+    gas,
+    storageLimit,
+    // currentNetwork.netId,
+    Boolean(provider),
+    Object.keys(tokensAmount)?.[0],
+    type,
+  ])
 
   if (loading) {
     return {loading}
   }
 
   if (error) {
+    console.log('error', error)
```

Comment:
better to  remove console

Reply:
it is used for debugging. when release I will remove the console.